src/aiProviders/continue/continueContextProvider.ts

```diff
@@ -0,0 +1,173 @@
+        .execute(10000);
+      if (result.success) {
+        const data: SelfCodeNode[] = result.data.map((row) => ({
+          ...row,
+          INITIAL_STACK: JSON.parse(row.INITIAL_STACK as unknown as string),
+        }));
+
+        return data;
+      }
+    } catch (e) {
+      vscode.window.showErrorMessage(
+        `An error occured fetching SELF code errors, and therefore will be disabled.`
+      );
+    }
+  }
+
+  async getContextItems(
+    query: string,
+    extras: ContextProviderExtras
+  ): Promise<ContextItem[]> {
+    const job: JobInfo = this.getCurrentJob();
+    const schema = this.getDefaultSchema();
+    const fullInput = extras.fullInput;
+    const contextItems: ContextItem[] = [];
+    contextItems.push({
+      name: `SYSTEM PROMPT`,
+      description: `system prompt context`,
+      content: DB2_SYSTEM_PROMPT,
+    });
+    try {
+      switch (true) {
+        case fullInput.includes(`*SELF`) || query?.includes(`*SELF`):
+          // get current self code errors in job
+          // build promt with error information
+          // add to contextItems
+
+          if (job) {
+            const selfCodes = await this.getSelfCodes(job);
+
+            let prompt = DB2_SELF_PROMPT.join(" ");
+            prompt += JSON.stringify(selfCodes, null, 2);
+
+            contextItems.push({
+              name: `${job.name}-self`,
+              description: `SELF code errors for ${job.name}`,
+              content: prompt,
+            });
+          }
+
+          return contextItems;
+        default:
+          // const contextItems: ContextItem[] = [];
+          const tableRefs = await findPossibleTables(
+            null,
+            schema,
+            fullInput.split(` `)
+          );
+          for (const table of Object.keys(tableRefs)) {
+            const columnData: TableColumn[] = tableRefs[table];
+            const tableSchema =
+              columnData.length > 0 ? columnData[0].TABLE_SCHEMA : null;
+
+            // create context item
+            let prompt = `Db2 for i Table meta data for schema ${tableSchema} table ${table}\n`;
+            prompt += `Column Info: ${JSON.stringify(columnData)}\n\n`;
+
+            contextItems.push({
+              name: `${job.name}-${tableSchema}-${table}`,
+              description: `Schema and table information for ${table}`,
+              content: prompt,
+            });
+          }
+
+          return contextItems;
+      }
+    } catch (error) {
+      vscode.window.showErrorMessage(`Failed to query Db2i database: ${error}`);
+      throw new Error(`Failed to query Db2i database: ${error}`);
+    } finally {
+    }
+  }
+  async loadSubmenuItems(
+    args: LoadSubmenuItemsArgs
+  ): Promise<ContextSubmenuItem[]> {
+    return [];
+  }
+}
+
+export async function registerContinueProvider() {
+  const provider = new db2ContextProvider();
+  const continueID = `Continue.continue`;
+  const continueEx = vscode.extensions.getExtension(continueID);
```

Comment:
`continueEx` can be undefined if the extension is not installed.

Also, you need to check if it's active or activate it if not.

Like this: https://github.com/codefori/vscode-db2i/blob/main/src/extension.ts#L109

Reply:
Also, if the extension is not installed, then all the new Continue specific commands need to be disabled.